src/plugins/core.js

```diff
@@ -137,44 +134,9 @@ function Plugin(options = {}) {
     const nextText = next.startText
     const nextChars = nextText.getDecorations(decorators)
```

Comment:
I think `nextChars`, `nextText` (and probably others) are now unused so we don't need to compute them too? There might be a handful more lines of code we can remove here which will improve performance when checking further.